File: backend/src/main/java/com/example/isabloodbank/model/Appointment.java
package com.example.isabloodbank.model;

import com.example.isabloodbank.model.enums.AppointmentStatus;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDate;
import java.time.LocalTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@Table(name = "appointment")
public class Appointment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private LocalDate scheduledTime;

    @Column
    private Integer duration;

    @OneToMany
    private List<User> assignedStaff = new ArrayList<>();

    @OneToOne
    @JoinColumn(name="donator_id")
    private User donator;

    @Column
    private boolean isReserved;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "work_calendar_id")
    private WorkCalendar workCalendar;

    private String description;

    private AppointmentStatus status;

    @Column
    private LocalTime startTime;
}

File: frontend/isa-blood-bank/src/app/components/work-calendars/shared/appointment.ts
import { User } from "../../users/shared/user.model";

export class Appointment {
    scheduledTime: string;
    duration: number;
    reserved: boolean;
    startTime: string;
    id: number;
    donator: User;

    constructor(scheduledTime: string, duration: number, reserved: boolean, startTime: string, 
            donator: User, id: number) {
        this.scheduledTime = scheduledTime;
        this.duration = duration;
        this.reserved = reserved;
        this.startTime = startTime;
        this.id = id;
        this.donator = donator;
    }
}